lib/client/db/oracle/oracle.go

```diff
@@ -86,6 +87,11 @@ func createClientWallet(signer crypto.Signer, certPem []byte, password string, w
 }
 
 func createJKSWallet(signer crypto.Signer, certPEM, caPEM []byte, password string) ([]byte, error) {
+	// unwrap *keys.PrivateKey if necessary.
+	if pk, ok := signer.(*keys.PrivateKey); ok {
+		signer = pk.Signer
+	}
```

Comment:
LGTM

but since `api/utils/key` is imported, could we change `signer crypto.Signer,` to `signer *keys.PrivateKey` in this file?

maybe move `x509.MarshalPKCS8PrivateKey` to `keys.PrivateKey` too as a helper

Reply:
Hmm, there are quite a few options here if we want to refactor this further.

@nklaassen any ideas here? Looks like these changes are due to:
- https://github.com/gravitational/teleport/pull/45995

Reply:
Sounds good, I've added `MarshalSoftwarePrivateKeyPKCS8DER` now. PTAL.